src/pages/app.js

```diff
@@ -0,0 +1,16 @@
+import React from "react"
+import { Router as MyRouter } from "@reach/router"
+import Login from '../components/Login'
+import PrivateRoute from "../components/PrivateRoute"
+import PrivateTest from "../components/privatetest"
+
+const Router = () => {
+  return (
+    <MyRouter>
+      <Login path="app/login" />
+      <PrivateRoute path="app/privatetest" component={PrivateTest} />
```

Comment:
je zou op het moment van inloggen de huidige route van de user in de state kunnen opslaan, dan kan je die state hier weer opvragen. Misschien is het zelfs mogelijk om de vorige route uit de location.history te halen